crates/subspace-node/src/secondary_chain/cli.rs

```diff
@@ -49,8 +51,8 @@ pub struct DomainCli {
     pub run_system: SubstrateRunCmd,
 
     /// Optional relayer address to relay messages on behalf.
-    #[arg(long)]
-    pub relayer_id: Option<String>,
+    #[clap(long, value_parser = parse_relayer_id)]
```

Comment:
Why not just use `value_parser = parse_relayer_id`, which has much less code and can tell the specific error when it really happens. `RelayerParseError` is not really beneficial IMHO.

```rust
fn parse_relay_id(s: &str) -> std::result::Result<RelayerId, sp_core::crypto::PublicError> {
    RelayerId::from_ss58check(s)
}
```